gui/src/app/Scripting/Analysis/AnalysisRWindow.tsx

```diff
@@ -28,19 +28,14 @@ const AnalysisRWindow: FunctionComponent<AnalysisWindowProps> = ({
     notRunnableReason,
   } = useAnalysisState(latestRun);
 
+  const { run } = useWebR({ consoleRef, imagesRef, onStatus });
```

Comment:
Looking inside the custom hook, Jeremy is right that you always depend on the individual components of the object rather than the object itself. So this should be fine not memoized. But I'd want to double-check just to be sure.

Reply:
As long as consoleRef, imagesRef, onStatus are being received by the hook individually, then you definitely do not need to memoize the args object. It's just a way of passing in individual args.